### ISSUE TITLE
Make modular squids

### ISSUE THREAD

**Issue by bdmason** [litentry/squid#42]
# Modular Squids

## Overview

Indexing balances is extremely slow as it requires `treasury.Deposit` to get fees (which reduce balances) and this is emitted for every single extrinsic. The multi chain squid isn't an issue as it runs each chain's processor in parallel. The solution is to remove the `SubstrateAccount` model (this can be replaced with an extension to the schema, it's effectively just something the other models point to via a foreign key). After this we can have 1 squid per data module (e.g. balances, governance, crowdloans etc). As most are quick to re-index it means that future work can be deployed quickly most of the time. The balances module, once finalised, will rarely (if ever) be updated, so the incredibly slow re-indexing process won't be an issue.

Each squid/data module has its own database (as it needs wiping for re-indexing) and graphql schema. The end result (due to schema stitching in Litentry Graph) will be a graph exactly as it is now.

### Sharable:

- /chains
- /data (nginx conf will just grow)
- /src/types
- /src/utils
- docker-compose.nginx.yml
- init-letsencrypt.sh (letsencrypt needs *.squid.litentry.io (we use module name as sub-sub domain))
Makefile (debatable, might make sense to make it modular and keep the original for shared commands e.g. exploreKusama)

### Needs to become modular:

- /db/{module} (migrations)
- /lib/{module}
- /src/model/{module}
- /src/processors/{module}/{chain}Processor.ts
- /src/handlers/{module}
- Dockerfile.{module}
- docker-compose.{module}.yml
- schema.{module}.graphql

### Ignore for now

- /logs

## Deployments:

We need a single nginx container, but the reloading of this triggers go live, we need to limit the going live to the completed module processors somehow... maybe we just need to block parallel module deployments... not too much of an issue as most will be quick enough
